aroquev/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -55,13 +67,16 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // TODO: verify that message is not empty
+    String username = request.getParameter("comments-username-input");
     String comment = request.getParameter("comments-body-input");
     long timestamp = System.currentTimeMillis();
 
     Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("commentUsername", username);
     commentEntity.setProperty("commentBody", comment);
     commentEntity.setProperty("timestamp", timestamp);
+    commentEntity.setProperty("upvotes", 0);
+    commentEntity.setProperty("downvotes", 0);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
```

Comment:
May not need redirect now?

Reply:
This redirect? 
`response.sendRedirect("/");`

Reply:
Correct. Most times I've seen POST request handlers (that create a new thing) return the thing that was created.